lib/mayaUsd/nodes/proxyShapeBase.cpp

```diff
@@ -746,6 +747,37 @@ MayaUsdProxyShapeBase::isStageValid() const
     return true;
 }
 
+MStatus
+MayaUsdProxyShapeBase::preEvaluation(const MDGContext& context, const MEvaluationNode& evaluationNode)
+{
+    if (evaluationNode.dirtyPlugExists(excludePrimPathsAttr)) {
+        _IncreaseExcludePrimPathsVersion();
+    }
+    else if (evaluationNode.dirtyPlugExists(outStageDataAttr) ||
+        // All the plugs that affect outStageDataAttr
+        evaluationNode.dirtyPlugExists(filePathAttr) ||
+        evaluationNode.dirtyPlugExists(primPathAttr) ||
+        evaluationNode.dirtyPlugExists(loadPayloadsAttr) ||
+        evaluationNode.dirtyPlugExists(inStageDataAttr) ||
+        evaluationNode.dirtyPlugExists(inStageDataCachedAttr)) {
+        _IncreaseUsdStageVersion();
+    }
+
+    return MStatus::kSuccess;
+}
+
+MStatus
+MayaUsdProxyShapeBase::postEvaluation(const  MDGContext& context, const MEvaluationNode& evaluationNode, PostEvaluationType evalType)
+{
+    // If/when the MPxDrawOverride for the proxy shape specifies
+    // isAlwaysDirty=false to improve performance, we must be sure to notify
+    // the Maya renderer that the geometry is dirty and needs to be redrawn
+    // when any plug on the proxy shape is dirtied.
+    MHWRender::MRenderer::setGeometryDrawDirty(thisMObject());
```

Comment:
I'm not sure why we need this? Some attributes (like input time) have `setAffectsAppearance` which I believe is what was causing the refresh.

Reply:
I don't know why we need it either which is why I don't want to remove it. Not sure which workflows need testing. setDependentsDirty has it and if the node goes into EM setDependentsDirty won't get called, so this is more correct. I think probably we need to figure out when exactly we need to do the refreshes but the change is large enough already I didn't want to add that investigation to it.

Reply:
The flip side is - nobody will know why it was added. It's a rather simple change to make, so let's wait for repro case.

Reply:
I still prefer it there. If you want I will comment out the call and leave a comment explaining why I added it.

Reply:
Documenting why we think it should be there (i.e. symmetry with setDependentsDirty) and waiting for confirmation before we enable the code sounds reasonable to me.